app/code/Magento/InventoryShipping/Model/PriorityShippingAlgorithm.php

```diff
@@ -0,0 +1,279 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\InventoryShipping\Model;
+
+use Magento\Framework\Api\SearchCriteriaBuilder;
+use Magento\InventoryApi\Api\Data\SourceInterface;
+use Magento\InventoryApi\Api\Data\SourceItemInterface;
+use Magento\InventoryApi\Api\Data\SourceItemInterfaceFactory;
+use Magento\InventoryApi\Api\GetAssignedSourcesForStockInterface;
+use Magento\InventoryApi\Api\SourceItemRepositoryInterface;
+use Magento\InventorySalesApi\Api\Data\SalesChannelInterface;
+use Magento\InventorySalesApi\Api\StockResolverInterface;
+use Magento\InventoryShipping\Model\ShippingAlgorithmResult\ShippingAlgorithmResultInterface;
+use Magento\InventoryShipping\Model\ShippingAlgorithmResult\ShippingAlgorithmResultInterfaceFactory;
+use Magento\InventoryShipping\Model\ShippingAlgorithmResult\SourceItemSelectionInterface;
+use Magento\InventoryShipping\Model\ShippingAlgorithmResult\SourceItemSelectionInterfaceFactory;
+use Magento\InventoryShipping\Model\ShippingAlgorithmResult\SourceSelectionInterface;
+use Magento\InventoryShipping\Model\ShippingAlgorithmResult\SourceSelectionInterfaceFactory;
+use Magento\Sales\Api\Data\OrderInterface;
+use Magento\Sales\Api\Data\OrderItemInterface;
+use Magento\Sales\Model\Order\Item as OrderItem;
+use Magento\Store\Api\WebsiteRepositoryInterface;
+use Magento\Store\Model\StoreManagerInterface;
+
+/**
+ * {@inheritdoc}
+ * This shipping algorithm just iterates over all the sources one by one in priority order
+ *
+ * @SuppressWarnings(PHPMD.CouplingBetweenObjects)
```

Comment:
We should not add suppressing in new code (only in special cases)
We can split this class on a few, first is loading data, second is perform algorithm